ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelper.java

```diff
@@ -177,4 +168,10 @@ private void disconnect() throws XAException {
         }
     }
 
+    private void assertIfRecoveryScanWasStarted() throws XAException {
+        if (delegate == null) {
+            throw new XAException("Recovery scan has to be started");
```

Comment:
The spec does say what code new XAException("") uses so you need to be throwing one of the valid XA codes, I would guess you either need XAER_RMERR or one of the protocol violation errors if the error is recoverable.